React/Views/RCTScrollView.m

```diff
@@ -358,6 +359,23 @@ - (UIView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event
   return [super hitTest:point withEvent:event];
 }
 
+- (BOOL)gestureRecognizerShouldBegin:(UIGestureRecognizer *)gestureRecognizer {
+  // ScrollView enables pinch gesture late in its lifecycle. So simply setting it
+  // in the setter gets overriden when the view loads. Revoke & disable gesture
+  // at first use.
+  if (!_pinchEnabled && [gestureRecognizer isKindOfClass:[UIPinchGestureRecognizer class]]) {
```

Comment:
And, also, if we could not find a way to disable `pinchGestureRecognizer`, we can directly compare `pinchGestureRecognizer` and `gestureRecognizer` here instead of using `isKindOfClass`.